cibuildwheel/macos.py

```diff
@@ -34,21 +37,63 @@ def call(args: Sequence[PathOrStr], env: Optional[Dict[str, str]] = None, cwd: O
     return subprocess.check_call(args, env=env, cwd=cwd, shell=shell)
 
 
+def get_macos_version() -> Tuple[int, int]:
+    '''
+    Returns the macOS major/minor version, as a tuple, e.g. (10, 15) or (11, 0)
+
+    These tuples can be used in comparisons, e.g.
+        (10, 14) <= (11, 0) == True
+        (10, 14) <= (10, 16) == True
+        (11, 2) <= (11, 0) != True
+    '''
+    version_str, _, _ = platform.mac_ver()
+    version = tuple(map(int, version_str.split(".")[:2]))
+    return cast(Tuple[int, int], version)
+
+
 class PythonConfiguration(NamedTuple):
     version: str
     identifier: str
     url: str
 
 
-def get_python_configurations(
-        build_selector: BuildSelector) -> List[PythonConfiguration]:
+def get_python_configurations(build_selector: BuildSelector,
+                              architectures: Set[Architecture]) -> List[PythonConfiguration]:
 
     full_python_configs = read_python_configs('macos')
 
     python_configurations = [PythonConfiguration(**item) for item in full_python_configs]
 
-    # skip builds as required
-    return [c for c in python_configurations if build_selector(c.identifier)]
+    # filter out configs that don't match any of the selected architectures
+    python_configurations = [c for c in python_configurations
+                             if any(c.identifier.endswith(a.value) for a in architectures)]
+
+    # skip builds as required by BUILD/SKIP
+    python_configurations = [c for c in python_configurations if build_selector(c.identifier)]
+
+    # When running on macOS 11 and x86_64, the reported OS is '10.16', but
+    # there is no such OS - it really means macOS 11.
+    if get_macos_version() >= (10, 16):
+        if any(c.identifier.startswith('pp') for c in python_configurations):
+            # pypy doesn't work on macOS 11 yet
+            # See https://foss.heptapod.net/pypy/pypy/-/issues/3314
+            log.warning(unwrap('''
```

Comment:
Should this be a warning or an error? Is someone really going to go the the logs and look for these warnings?

Reply:
GHA pulls them up to the summary screen as 'annotations', but I take your point. Trouble is, if this is an error then we can't run without an explicit `CIBW_SKIP=pp*-macosx*`, which is a very strange first-run journey / minimal config.

Reply:
Yes, but it means if you simply upgrade from macOS-10.15 to macOS-11.0, suddenly your PyPy wheels disappear. :/ If you were not building them before, that's not a big deal, but if you were trying to build them and they go away with a green checkmark still, that seems problematic.

Reply:
Yeah, fair point. Now that we can recommend people stick with 10.15, I'm happy to upgrade this to an error.

Reply:
Ah.. actually this would mean that every integration test would need to add `CIBW_SKIP: pp*-macosx*` so the suite passes on macOS-11.... maybe that's okay, because we're not testing on macOS-11 yet? 

Looks like theres some progress on PyPy Big Sur support, so maybe it'll be resolved before people start trying to run on macOS-11. Otherwise we might have to switch it back to a warning again, just to save our sanity ;)

Reply:
Yeah... actually, raising an error here means I can't run unit tests locally, because `get_python_configurations` is called by `main`... I'm leaning towards keeping this a warning, and hopefully PyPy have a macOS 11 release out before macOS 11 is mainstream in CIs.

Reply:
Okay, in that case, for now, it can be a warning. If macos-latest starts looking like it will be 11.0 and PyPy doesn't support it yet, then we'll see about an error?

Reply:
Yeah, sounds good, we'll pick up the conversation then if it becomes an issue.